### PR TITLE
Add ios sample

### PR DESCRIPTION
**Pull request by naoki-akimoto**
iOSのサンプルアプリを追加しました。レビューをお願いします。

手元にてsnippet.shとbase.mdを使って各タグが問題ないことは確認しています。

### CODE REVIEW

**Comment by satoshikumano** [comment]
@naoki-akimoto @yongpingchen スニペットの内容は任せます。

**Review by satoshikumano — APPROVED** [review]

**Review by yongpingchen — CHANGES_REQUESTED** [review]
@naoki-akimoto please add mark down template file.

File: snippets/ios/SampleProject/SampleProject/ViewController.swift [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+import UIKit
+import ThingIF
+
+class ViewController: UIViewController {
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Do any additional setup after loading the view, typically from a nib.
+
+        let owner = Owner(TypedID(.thing, id: "dummyID"), accessToken: "dummyToken")
```
Comment (yongpingchen) [inline snippets/ios/SampleProject/SampleProject/ViewController.swift:16]: `owner` should be in `variables`?

File: snippets/ios/SampleProject/SampleProject/ViewController.swift [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+        // CodeTagStart: variables
```
Comment (yongpingchen) [inline snippets/ios/SampleProject/SampleProject/ViewController.swift]: tag name should be different with other platform, you can use like `variables_ios`

File: snippets/ios/SampleProject/SampleProject/ViewController.swift [hunk 3]
```diff
@@ -0,0 +1,417 @@
...
+        // CodeTagEnd: variables
+
+        // 1. Create ThingIFAPI
```
Comment (yongpingchen) [inline snippets/ios/SampleProject/SampleProject/ViewController.swift]: tag name should be different with other platforms

File: snippets/ios/SampleProject/SampleProject/ViewController.swift [hunk 4]
```diff
@@ -0,0 +1,417 @@
...
+        // CodeTagStart: thingIFAPI
+        let app = KiiApp("___APPID___", appKey: "___APPKEY___", site: .jp)
+        let api = ThingIFAPI(app, owner: owner)
+        // CodeTagEnd: thingIFAPI
+
+        // 2. Execute Onboard
+        do {
+            // CodeTagStart: onboard
```
Comment (yongpingchen) [inline snippets/ios/SampleProject/SampleProject/ViewController.swift]: tag name should be different with other platforms

File: snippets/ios/SampleProject/SampleProject/ViewController.swift [hunk 5]
```diff
@@ -0,0 +1,417 @@
...
+                            break
+                        default:
+                            break
+                        }
+                    }
+            }
+            // CodeTagEnd: onboard
+        }
+
+        // CodeTagStart: updateThingType
```
Comment (yongpingchen) [inline snippets/ios/SampleProject/SampleProject/ViewController.swift]: tag name should be different with other platforms

File: snippets/ios/SampleProject/SampleProject/ViewController.swift [hunk 6]
```diff
@@ -0,0 +1,417 @@
...
+                    }
+                    break
+                default:
+                    break
+                }
+            }
+        }
+        // CodeTagEnd: updateThingType
+
+        // CodeTagStart: updateFirmwareVersion
```
Comment (yongpingchen) [inline snippets/ios/SampleProject/SampleProject/ViewController.swift]: tag name should be different with other platforms

File: snippets/ios/SampleProject/SampleProject/ViewController.swift [hunk 7]
```diff
@@ -0,0 +1,417 @@
...
+                    default:
+                        break
+                    }
+                    break
+                default:
+                    break
+                }
+            }
+        }
+        // CodeTagEnd: updateFirmwareVersion
```
Comment (yongpingchen) [inline snippets/ios/SampleProject/SampleProject/ViewController.swift]: tag name should be different with other platforms

File: snippets/ios/SampleProject/SampleProject/ViewController.swift [hunk 8]
```diff
@@ -0,0 +1,417 @@
...
+
+        do {
+            // CodeTagStart: postNewCommand
```
Comment (yongpingchen) [inline snippets/ios/SampleProject/SampleProject/ViewController.swift]: tag name should be different with other platforms

File: snippets/ios/SampleProject/SampleProject/ViewController.swift [hunk 9]
```diff
@@ -0,0 +1,417 @@
...
+                        break
+                    default:
+                        break
+                    }
+                }
+            }
+            // CodeTagEnd: postNewCommand
+        }
+
+        // CodeTagStart: getCommand
```
Comment (yongpingchen) [inline snippets/ios/SampleProject/SampleProject/ViewController.swift]: tag name should be different with other platforms

File: snippets/ios/SampleProject/SampleProject/ViewController.swift [hunk 10]
```diff
@@ -0,0 +1,417 @@
...
+                    let data = result.data
+                    let succeeded = result.succeeded
+                    let errorMessage = result.errorMessage
+                }
+            }
+        }
+        // CodeTagEnd: getCommand
+
+        do {
+            // CodeTagStart: postNewTrigger
```
Comment (yongpingchen) [inline snippets/ios/SampleProject/SampleProject/ViewController.swift]: tag name should be different with other platforms

File: snippets/ios/SampleProject/SampleProject/ViewController.swift [hunk 11]
```diff
@@ -0,0 +1,417 @@
...
+                    }
+                } else {
+                    // check trigger.
+                }
+            }
+            // CodeTagEnd: postNewTrigger
+        }
+
+        do {
+            // CodeTagStart: patchTrigger
```
Comment (yongpingchen) [inline snippets/ios/SampleProject/SampleProject/ViewController.swift]: tag name should be different with other platforms

File: snippets/ios/SampleProject/SampleProject/ViewController.swift [hunk 12]
```diff
@@ -0,0 +1,417 @@
...
+                    }
+                } else {
+                    // check trigger.
+                }
+            }
+            // CodeTagEnd: patchTrigger
+        }
+
+        do {
+            // CodeTagStart: ungroupedQuery
```
Comment (yongpingchen) [inline snippets/ios/SampleProject/SampleProject/ViewController.swift]: tag name should be different with other platforms

File: snippets/ios/SampleProject/SampleProject/ViewController.swift [hunk 13]
```diff
@@ -0,0 +1,417 @@
...
+                    for state in states! {
+                        // check HistoryState.
+                    }
+                }
+            }
+            // CodeTagEnd: ungroupedQuery
+        }
+
+        do {
+            // CodeTagStart: groupedQuery
```
Comment (yongpingchen) [inline snippets/ios/SampleProject/SampleProject/ViewController.swift]: tag name should be different with other platforms

File: snippets/ios/SampleProject/SampleProject/ViewController.swift [hunk 14]
```diff
@@ -0,0 +1,417 @@
...
+                    for states in statesArray! {
+                        // check GroupedHistoryStates.
+                    }
+                }
+            }
+            // CodeTagEnd: groupedQuery
+        }
+
+        do {
+            // CodeTagStart: aggregate
```
Comment (yongpingchen) [inline snippets/ios/SampleProject/SampleProject/ViewController.swift]: tag name should be different with other platforms

File: snippets/ios/SampleProject/SampleProject/ViewController.swift (hunk 1 above)
Comment (naoki-akimoto) [reply snippets/ios/SampleProject/SampleProject/ViewController.swift:16]: In this sample, no values are in `variables`
If I insert `owner`, it's `thingIFAPI(_ios)` like this,
```
let app=**
let owner=**
let api=ThingIFAPI(app, owner: owner)
```

Comment (naoki-akimoto) [reply snippets/ios/SampleProject/SampleProject/ViewController.swift:16]: reference: http://docs.kii.com/en/guides/thingifsdk/ios/initialize/

**Comment by naoki-akimoto** [comment]
@yongpingchen fixed. Please confirm.

**Review by yongpingchen — APPROVED** [review]

**Pull request merged**